### DISCUSSION TITLE
[Quasar CLI Vite] No built-in type checking?

### DISCUSSION THREAD

**Discussion by philfontaine** [quasarframework/quasar#12821] [Ideas / Proposals] [answered]
Vite does not do type checking to speed up development.

For the build script, I'd like to typecheck my code, as shown here https://vitejs.dev/guide/features.html#typescript

I can adjust the script myself and find the right commands/options, but I thought this is something Quasar might want to have built-in when running the build command.

EDIT: I now realize that typechecking was not done in the webpack CLI either, so maybe this was never meant to be part of quasar.

**Comment by rstoenescu** [comment, marked as answer]
Hi,

Yes, just like Vite also recommends, you can configure the IDE instead to do typechecking OR use vue-tsc package when building for production.
We'll be investing some time into a baked-in solution (which uses vue-tsc) in the future though.

**Reply by soulsam480** [reply]
Hii @rstoenescu , Is it possible for us to bake in https://github.com/fi3ework/vite-plugin-checker as the checker ? This can substitute `fork-ts-checker-webpack-plugin` as used in quasar/app-webpack ?

**Reply by soulsam480** [reply]
Also it's pretty comprehensive and also runs inside workers. It'd be really good if you consider to give it as an option. Thx

**Reply by rstoenescu** [reply]
> Also it's pretty comprehensive and also runs inside workers. It'd be really good if you consider to give it as an option. Thx
> 
> 

Will give some thought to it as soon as I have some free bandwidth. Meanwhile, I see they also have a vite plugin, so you can already play with it (quasar.config.js > build > vitePlugins). You might want to disable the current linter though (comment out quasar.config.js > eslint).
